src/main/java/org/springframework/data/couchbase/core/convert/MappingCouchbaseConverter.java

```diff
@@ -501,33 +507,33 @@ public void doWithPersistentProperty(final CouchbasePersistentProperty prop) {
 				}
 
 				Object propertyObj = accessor.getProperty(prop, prop.getType());
-				if (null != propertyObj) {
-					if (prop.isAnnotationPresent(IdPrefix.class)) {
-						IdPrefix prefix = prop.findAnnotation(IdPrefix.class);
-						int order = prefix.order();
-						prefixes.put(order, convertToString(propertyObj));
-						return;
-					}
-
-					if (prop.isAnnotationPresent(IdSuffix.class)) {
-						IdSuffix suffix = prop.findAnnotation(IdSuffix.class);
-						int order = suffix.order();
-						suffixes.put(order, convertToString(propertyObj));
-						return;
-					}
-
-					if (prop.isAnnotationPresent(IdAttribute.class)) {
-						IdAttribute idAttribute = prop.findAnnotation(IdAttribute.class);
-						int order = idAttribute.order();
-						idAttributes.put(order, convertToString(propertyObj));
-					}
-
-					if (!conversions.isSimpleType(propertyObj.getClass())) {
-						writePropertyInternal(propertyObj, target, prop);
-					} else {
-						writeSimpleInternal(propertyObj, target, prop.getFieldName());
-					}
+
+				if (prop.isAnnotationPresent(IdPrefix.class)) {
+					IdPrefix prefix = prop.findAnnotation(IdPrefix.class);
+					int order = prefix.order();
+					prefixes.put(order, convertToString(propertyObj));
+					return;
+				}
+
+				if (prop.isAnnotationPresent(IdSuffix.class)) {
+					IdSuffix suffix = prop.findAnnotation(IdSuffix.class);
+					int order = suffix.order();
+					suffixes.put(order, convertToString(propertyObj));
+					return;
+				}
+
+				if (prop.isAnnotationPresent(IdAttribute.class)) {
+					IdAttribute idAttribute = prop.findAnnotation(IdAttribute.class);
+					int order = idAttribute.order();
+					idAttributes.put(order, convertToString(propertyObj));
 				}
+
+				if (propertyObj != null && !conversions.isSimpleType(propertyObj.getClass())) {
```

Comment:
Nullability-annotations can be tricky. There are JSR-305 and Spring Framework variants of which Spring's annotations are meta-annotated with the JSR-305 ones. We have already nullability utilities in place (see [`o.s.d.util.NullableUtils`](https://github.com/spring-projects/spring-data-commons/blob/master/src/main/java/org/springframework/data/util/NullableUtils.java).

The nullability lookup should also go into `CouchbasePersistentProperty`. We should consider extending `NullableUtils` for now to check whether a property is explicitly marked as nullable. Right now, the utilities only consider `MethodParameter` which is applicable for property accessors but not fields.

Reply:
I have an isNullable() in CouchbasePersistentProperty now.  It is referenced in the 'else'.   (I think I need to re-push).

				if (propertyObj != null && !conversions.isSimpleType(propertyObj.getClass())) {
					writePropertyInternal(propertyObj, target, prop);
				} else if (propertyObj != null || prop.isNullable()) {
					writeSimpleInternal(propertyObj, target, prop.getFieldName());
				}